wsd/DocumentBroker.cpp

```diff
@@ -1121,13 +1121,15 @@ bool DocumentBroker::download(const std::shared_ptr<ClientSession>& session, con
                     if (inputs != 1 || outputs != 1)
                         throw std::exception();
 
-                    int process = Util::spawnProcess(command, args);
-                    int status = -1;
-                    const int rc = ::waitpid(process, &status, 0);
-                    if (rc != 0)
-                    {
-                        LOG_ERR("Conversion from " << extension << " to " << newExtension << " failed (" << rc << ").");
-                        return false;
+                    if (!Util::isKitInProcess()) {
+                        int process = Util::spawnProcess(command, args);
+                        int status = -1;
+                        const int rc = ::waitpid(process, &status, 0);
+                        if (rc != 0)
+                        {
+                            LOG_ERR("Conversion from " << extension << " to " << newExtension << " failed (" << rc << ").");
+                            return false;
+                        }
```

Comment:
Command used to be a separate process. It has a main function. Probably you need an else branch here, so that you invoke that main function on a thread?